File: src/main/java/com/example/the_machine/dto/ToolResultContentDto.java
package com.example.the_machine.dto;

import com.fasterxml.jackson.databind.JsonNode;

public record ToolResultContentDto(
    String name,
    JsonNode result,
    String callId
) implements ContentBlockDto {

  @Override
  public String type() {
    return ContentBlockTypes.TOOL_RESULT;
  }
}

File: src/main/kotlin/com/example/the_machine/dto/content/ToolResultContentDto.kt
package com.example.the_machine.dto.content

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Serializable
@SerialName(ContentBlockTypes.TOOL_RESULT)
data class ToolResultContentDto(
  val name: String,
  val result: JsonElement,
  val callId: String
) : ContentBlockDto